static/js/containers/ProfilePage.js

```diff
@@ -40,9 +39,7 @@ class ProfilePage extends React.Component {
       dispatch(startProfileEdit());
     }
     return dispatch(validateProfile(profile, requiredFields, messages)).then(() => {
-      dispatch(saveProfile(SETTINGS.username, profile)).then(() => {
-        dispatch(clearProfileEdit());
-      });
+      dispatch(saveProfile(SETTINGS.username, profile));
```

Comment:
Sorry, I guess I'm not quite understanding what the problem is. I pulled down your code and the problem seems to me to be that when we run the `requireCompleteProfile` method (in `App.js`, called on `componentDidMount` and `componentDidUpdate`) we're still not dispatching the `startProfileEdit` action - I don't understand exactly why that's not happening.

We shouldn't need to make any modification of the code in this method here (`saveProfile`) since we won't be running it when we load the tab.

Does that make sense? This code here (where we dispatch the `clearProfileEdit` action) is only run when we save a tab (click on the 'save and continue' button or w/e).